src/ape/types/manifest.py

```diff
@@ -47,13 +47,28 @@ class PackageManifest(FileMixin, SerializableType):
     #       version as `manifest`
     buildDependencies: Optional[Dict[str, str]] = None
 
+    def __getattr__(self, attr_name: str):
+        if self.contractTypes and attr_name in self.contractTypes:
+            return self.contractTypes[attr_name]
+
+        else:
+            raise AttributeError(f"{self.__class__.__name__} has no attribute '{attr_name}'")
+
     def to_dict(self):
         data = super().to_dict()
 
         if "contractTypes" in data and data["contractTypes"]:
             for name in data["contractTypes"]:
                 # NOTE: This was inserted by us, remove it
                 del data["contractTypes"][name]["contractName"]
+                # convert Path to str, or else we can't serialize this as JSON
+                data["contractTypes"][name]["sourceId"] = str(
+                    data["contractTypes"][name]["sourceId"]
+                )
+                if "sourcePath" in data["contractTypes"][name]:
+                    data["contractTypes"][name]["sourcePath"] = str(
+                        data["contractTypes"][name]["sourcePath"]
+                    )
```

Comment:
Maybe this file path normalization logic should live in `src/ape/abstract.py` as a separate handling case for `Path` objects in `SerializableType`'s `to_dict` and `from_dict` methods

Reply:
Gave that a try and couldn't find an easy way to roundtrip `Path` objects somewhere in the nested structure. If you recursively convert all `Path`s to `str`s, then in the `from_dict` you don't have enough info to determine if an incoming string is a `Path` or not.

It might be possible to get that to work using type annotations, but I'd rather do it in a separate PR and land this.

Reply:
What if they were URIs, and you could check if a URI is in the string object.

Either that, or sublclass `from_dict` manually for those items